src/communication/types.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef COMMUNICATION_TYPES_HPP
+#define COMMUNICATION_TYPES_HPP
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include <aos/common/tools/error.hpp>
+
+#include "iamclient/types.hpp"
+
+namespace aos::mp::communication {
+
+class CommChannelItf {
+public:
+    ~CommChannelItf() = default;
```

Comment:
missing comment for the class, cosntructor & Connect method 
destructor should be virtual